### PR TITLE
Generate notification messages from associated message

### PR DESCRIPTION
**Pull request by elliotnash**
This pr fixes reaction notifications being in the wrong language when sent from a device in another language.

### CODE REVIEW

File: lib/helpers/message_helper.dart [hunk 1]
```diff
@@ -318,80 +290,76 @@ class MessageHelper {
...
+      if (associatedMessage != null) {
+        // grab the verb we'll use from the reactionToVerb map
+        String? verb = ReactionTypes.reactionToVerb[message.associatedMessageType];
+        // we need to check balloonBundleId first because for some reason
+        // game pigeon messages have the text "�"
+        if (associatedMessage.isInteractive()) {
+          return "$sender $verb ${getInteractiveText(associatedMessage)}";
+        // now we check if theres a subject or text and construct out message based off that
+        } else if (isNullOrEmpty(associatedMessage.text, trimString: true) == false
+            || isNullOrEmpty(associatedMessage.text, trimString: true) == false) {
```
Comment (tneotia) [inline lib/helpers/message_helper.dart:309]: This condition can be shortened to `!isNullOrEmpty(associatedMessage.text, trimString: true) || !isNullOrEmpty(associatedMessage.text, trimString: true)`

**Review by tneotia — CHANGES_REQUESTED** [review]
Minor change but otherwise great stuff! :)

File: lib/helpers/message_helper.dart (hunk 1 above)
Comment (elliotnash) [reply lib/helpers/message_helper.dart:309]: isNullOrEmpty returns an optional bool (not entirely sure why) so I'd have to unwrap the bool first, I'm not sure if its okay to assert isNullOrEmpty as not null

Comment (tneotia) [reply lib/helpers/message_helper.dart:309]: Ah I forgot about that, yes you can assert it as not null. It should never return null, but for some reason one of the packages we use has its own `nullOrEmpty` functions return null for whatever reason.

**Review by tneotia — APPROVED** [review]
LGTM :)

**Comment by zlshames** [comment]
Good work, thanks for the contribution!

**Pull request merged**